<a href="https://colab.research.google.com/github/KunalSharma2001/Data-Analysis-Projects/blob/main/NYC_Airbnb_DataAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
nyc_logs = pd.read_csv("/content/AB_NYC_2019.csv")
nyc_logs.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [ ]:
nyc_logs.tail()

id                                               name   host_id  \
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23

In [ ]:
nyc_logs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

# Insights
---
There are 15 columns and about 48894 rows in the given dataset.
According to the .info()
The NaN/Null values in the data are:
1. Name
2. Host-Name
3. Last-Review
4. Reviews-per-month

Reasons for the missing values:
1. The host or the person might have forgetten to enter the values in the dataset.
2. Hardware or Software Errors in the collecion may affect the data.

---
Therefore, we can either remove these rows or we can fill them with the mean values of the column.



# **Identifying and Replacing the Missing Values**

In [ ]:
print("Rows with missing values for Name")
print(nyc_logs['name'].isnull().value_counts())
print("\n")
print("Rows with missing value for Host_Name")
print(nyc_logs['host_name'].isnull().value_counts())
print("\n")
print("Rows with missing value for Last_Review")
print(nyc_logs['last_review'].isnull().value_counts())
print("\n")
print("Rows with missing values with Reviews_Per_Month")
print(nyc_logs['reviews_per_month'].isnull().value_counts())

Rows with missing values for Name
False    48879
True        16
Name: name, dtype: int64


Rows with missing value for Host_Name
False    48874
True        21
Name: host_name, dtype: int64


Rows with missing value for Last_Review
False    38843
True     10052
Name: last_review, dtype: int64


Rows with missing values with Reviews_Per_Month
False    38843
True     10052
Name: reviews_per_month, dtype: int64


# **Replacing Missing Values**

In [ ]:
nyc_logs['reviews_per_month'] = nyc_logs['reviews_per_month'].fillna(0.0)
nyc_logs.head(20)

id                                              name  host_id  \
0   2539                Clean & quiet apt home by the park     2787   
1   2595                             Skylit Midtown Castle     2845   
2   3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3   3831                   Cozy Entire Floor of Brownstone     4869   
4   5022  Entire Apt: Spacious Studio/Loft by central park     7192   
5   5099         Large Cozy 1 BR Apartment In Midtown East     7322   
6   5121                                   BlissArtsSpace!     7356   
7   5178                  Large Furnished Room Near B'way      8967   
8   5203                Cozy Clean Guest Room - Family Apt     7490   
9   5238                Cute & Cozy Lower East Side 1 bdrm     7549   
10  5295                  Beautiful 1br on Upper West Side     7702   
11  5441                   Central Manhattan/near Broadway     7989   
12  5803    Lovely Room 1, Garden, Best Area, Legal rental     9744   
13  6021  Wonderful Guest Bedroom in Manhattan for SINGLES    11528   
14  6090                     West Village Nest - Superhost    11975   
15  6848                  Only 2 stops to Manhattan studio    15991   
16  7097                 Perfect for Your Parents + Garden    17571   
17  7322                                   Chelsea Perfect    18946   
18  7726   Hip Historic Brownstone Apartment with Backyard    20950   
19  7750                 Huge 2 BR Upper East  Cental Park    17985   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
5              Chris           Manhattan         Murray Hill  40.74767   
6              Garon            Brooklyn  Bedford-Stuyvesant  40.68688   
7           Shunichi           Manhattan      Hell's Kitchen  40.76489   
8          MaryEllen           Manhattan     Upper West Side  40.80178   
9                Ben           Manhattan           Chinatown  40.71344   
10              Lena           Manhattan     Upper West Side  40.80316   
11              Kate           Manhattan      Hell's Kitchen  40.76076   
12            Laurie            Brooklyn         South Slope  40.66829   
13           Claudio           Manhattan     Upper West Side  40.79826   
14             Alina           Manhattan        West Village  40.73530   
15     Allen & Irina            Brooklyn        Williamsburg  40.70837   
16              Jane            Brooklyn         Fort Greene  40.69169   
17              Doti           Manhattan             Chelsea  40.74192   
18  Adam And Charity            Brooklyn       Crown Heights  40.67592   
19              Sing           Manhattan         East Harlem  40.79685   

    longitude        room_type  price  minimum_nights  number_of_reviews  \
0   -73.97237     Private room    149               1                  9   
1   -73.98377  Entire home/apt    225               1                 45   
2   -73.94190     Private room    150               3                  0   
3   -73.95976  Entire home/apt     89               1                270   
4   -73.94399  Entire home/apt     80              10                  9   
5   -73.97500  Entire home/apt    200               3                 74   
6   -73.95596     Private room     60              45                 49   
7   -73.98493     Private room     79               2                430   
8   -73.96723     Private room     79               2                118   
9   -73.99037  Entire home/apt    150               1                160   
10  -73.96545  Entire home/apt    135               5                 53   
11  -73.98867     Private room     85          

In [ ]:
print(nyc_logs['reviews_per_month'].isnull().value_counts())

False    48895
Name: reviews_per_month, dtype: int64


In [ ]:
nyc_logs['last_review'] = nyc_logs('last_review').fillna()